### PR TITLE
create and style signup screen, add new user at signup

### PR DESCRIPTION
**Pull request by juliabrowne**

### CODE REVIEW

File: js/components/SignupForm/SignupForm.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+  callingAPI = async values => {
+    const res = await getAPI('/auth/register', {
+      method: 'POST',
+      body: JSON.stringify(values)
+    })
+  }
+
+  onSubmit = values => {
+    console.log(values)
```
Comment (siddharthparmar7) [inline js/components/SignupForm/SignupForm.js]: remove all the console logs.

File: js/App.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import React, { Component } from "react";
 import AppNavigator from "./navigation/RootStackNavigator";
```
Comment (siddharthparmar7) [inline js/App.js]: remove all the unnecessary imports

File: js/components/SignupForm/SignupForm.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
+import React, { Component } from 'react'
+import {
+  View,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  Image,
+  ImageBackground,
+  ScrollView,
+  Switch
```
Comment (siddharthparmar7) [inline js/components/SignupForm/SignupForm.js]: remove all the unnecessary imports.

File: js/components/SignupForm/SignupForm.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+export default class SignupForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = { hidePassword: true }
+  }
+
+  callingAPI = async values => {
+    const res = await getAPI('/auth/register', {
+      method: 'POST',
+      body: JSON.stringify(values)
```
Comment (siddharthparmar7) [inline js/components/SignupForm/SignupForm.js:22]: const res is never used. Why?

File: js/components/SignupForm/SignupForm.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+                <Text style={styles.signupHeader}>First Name</Text>
+                <View style={styles.usernameInput}>
+                  <Field
+                    name='first_name'
+                    render={({ input, meta }) => {
+                      return (
+                        <TextInput
+                          {...input}
+                          style={{ fontFamily: 'Avenir' }}
+                        />
```
Comment (siddharthparmar7) [inline js/components/SignupForm/SignupForm.js:63]: error state and validation for your form? Please validate your inputs.

File: js/components/SignupForm/styles.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    ...globalStyles.inputBoxFont,
+    backgroundColor: '#FFFF'
+  },
+  togglePassword:
+  {
+    height: height / 20,
+    width: width / 14,
+    position: 'absolute',
+    right: 4,
+  },
```
Comment (siddharthparmar7) [inline js/components/SignupForm/styles.js:36]: use prettier to format your code.

File: js/screens/UserSignup/UserSignup.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import React, { Component, Fragment } from 'react'
+import { View, Image, ImageBackground, ScrollView } from 'react-native'
+import SignupForm from '../../components/SignupForm'
+import ChooseAvatar from '../../components/ChooseAvatar'
+import styles from './styles'
```
Comment (siddharthparmar7) [inline js/screens/UserSignup/UserSignup.js:5]: remove all the unnecessary imports.

File: js/screens/UserSignup/styles.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import { StyleSheet, Dimensions } from 'react-native'
+import globalStyles from '../../config/styles'
+const { width, height } = Dimensions.get('window')
```
Comment (siddharthparmar7) [inline js/screens/UserSignup/styles.js:2]: remove unnecessary imports.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please make sure you file names and components are semantic. If you have a component SingUp, then you should have a component SignIn, not LogIn. Please make the changes. Thank you.

File: js/components/SignupForm/SignupForm.js (hunk 1 above)
Comment (juliabrowne) [reply js/components/SignupForm/SignupForm.js]: done!

File: js/components/SignupForm/SignupForm.js (hunk 3 above)
Comment (juliabrowne) [reply js/components/SignupForm/SignupForm.js]: done!

File: js/components/SignupForm/SignupForm.js (hunk 5 above)
Comment (juliabrowne) [reply js/components/SignupForm/SignupForm.js:63]: done!

File: js/components/SignupForm/styles.js (hunk 6 above)
Comment (juliabrowne) [reply js/components/SignupForm/styles.js:36]: done!

File: js/screens/UserSignup/UserSignup.js (hunk 7 above)
Comment (juliabrowne) [reply js/screens/UserSignup/UserSignup.js:5]: done!

File: js/screens/UserSignup/styles.js (hunk 8 above)
Comment (juliabrowne) [reply js/screens/UserSignup/styles.js:2]: done!

File: js/components/SignupForm/SignupForm.js (hunk 4 above)
Comment (juliabrowne) [reply js/components/SignupForm/SignupForm.js:22]: it was originally there for a console log. i removed it!

File: js/App.js (hunk 2 above)
Comment (juliabrowne) [reply js/App.js]: done!

File: js/components/SignupForm/SignupForm.js (hunk 5 above)
Comment (siddharthparmar7) [reply js/components/SignupForm/SignupForm.js:63]: I don't see it there. Please make the changes.

File: js/components/SignupForm/SignupForm.js [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+
+                <Text style={styles.signupHeader}>Last Name</Text>
+                <View style={styles.usernameInput}>
+                  <Field
+                    name='last_name'
+                    render={({ input, meta }) => {
+                      return (
+                        <TextInput
+                          {...input}
+                          style={{ fontFamily: 'Avenir' }}
```
Comment (siddharthparmar7) [inline js/components/SignupForm/SignupForm.js:77]: show errors for all your text input.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please make the changes. Thank you.

File: js/components/SignupForm/SignupForm.js [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+              <View style={styles.inputContainer}>
+                <Text style={styles.signupHeader}>First Name</Text>
+                <View style={styles.usernameInput}>
+                  <Field
+                    name='first_name'
+                    render={({ input, meta }) => {
+                      return (
+                        <TextInput
+                          {...input}
+                          style={{ fontFamily: 'Avenir' }}
```
Comment (siddharthparmar7) [inline js/components/SignupForm/SignupForm.js:61]: move all of your styles to styles.js file.

File: js/components/SignupForm/SignupForm.js [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+
+                <Text style={styles.signupHeader}>Last Name</Text>
+                <View style={styles.usernameInput}>
+                  <Field
+                    name='last_name'
+                    render={({ input, meta }) => {
+                      return (
+                        <TextInput
+                          {...input}
+                          style={{ fontFamily: 'Avenir' }}
```
Comment (siddharthparmar7) [inline js/components/SignupForm/SignupForm.js:77]: move your styles to styles.js file.

File: js/components/SignupForm/SignupForm.js [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+                <View style={styles.usernameInput}>
+                  <Field
+                    name='email'
+                    render={({ input, meta }) => {
+                      return (
+                        <TextInput
+                          secureTextEntry={true}
+                          {...input}
+                          autoCapitalize='none'
+                          style={{ fontFamily: 'Avenir' }}
```
Comment (siddharthparmar7) [inline js/components/SignupForm/SignupForm.js:95]: move your styles to styles.js

File: js/components/SignupForm/SignupForm.js [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+                  <Field
+                    name='password'
+                    render={({ input, meta }) => {
+                      return (
+                        <View>
+                          <TextInput
+                            secureTextEntry={true}
+                            {...input}
+                            secureTextEntry={this.state.hidePassword}
+                            style={{ fontFamily: 'Avenir' }}
```
Comment (siddharthparmar7) [inline js/components/SignupForm/SignupForm.js:114]: move your styles to styles.js

File: js/components/SignupForm/SignupForm.js [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+                  <Field
+                    name='password_confirmation'
+                    render={({ input, meta }) => {
+                      return (
+                        <View>
+                          <TextInput
+                            secureTextEntry={true}
+                            {...input}
+                            secureTextEntry={this.state.hidePassword}
+                            style={{ fontFamily: 'Avenir' }}
```
Comment (siddharthparmar7) [inline js/components/SignupForm/SignupForm.js:148]: move your styles to styles.js

**Review by siddharthparmar7 — APPROVED** [review]
Make the changes in your next pull request. And make sure all of your styles are in styles.js file from now on to get your pull request merged into the develop branch. Thank you.

**Pull request merged**